src/views/tenants/tenants.jsx

```diff
@@ -1,140 +1,283 @@
+        setDisplayTenants(getDisplayTenants(sortedTenants, showHoused, showArchived));
+
+        const archivedTenants = sortedTenants.filter(tenant => tenant.archived);
+        setNonSelectableRows(archivedTenants.map(archivedTenant => archivedTenant.id))
+
       })
       .catch((error) => {
-        Toast(error.message, "error");
-        console.log(error);
+        Toast(error.message, "error")
       });
-  };
-
-  render() {
-    return (
-      <UserContext.Consumer>
-        {session => {
-          this.context = session;
-          return (
-            <div className='main-container'>
-              <div>
-                <div className="section-header">
-                  <h2 className="page-title">Tenants</h2>
-                  <Link className="button is-primary is-rounded ml-4" to="/add/tenant">+ ADD NEW</Link>
-                </div>
-                <div className="search-section">
-                  <Search placeholderMessage="Search tenants by name, property, or JOIN staff" />
-                </div>
-                <div className="properties-list">
-                  <BootstrapTable
-                    keyField='id'
-                    data={this.state.tenants}
-                    columns={columns}
-                    selectRow={selectRow}
-                    bootstrap4={true}
-                    headerClasses="table-header"
-                    classes="table-responsive"
-                  />
-                </div>
-              </div>
-            </div>
-          );
-        }}
-      </UserContext.Consumer>
-    );
   }
-}
+
+  const handleToggleHoused = () => {
+    const newShowHoused = !showHoused;
+    setShowHoused(newShowHoused)
+
+    if (isSearchActive) {
+      setDisplayTenants(getDisplayTenants(searchedTenants, !showHoused, showArchived));
+    } else {
+      setDisplayTenants(getDisplayTenants(allTenants, !showHoused, showArchived));
+    }
+    const newSelectedTentants = getDisplayTenants(selectedTenants, !showHoused, showArchived)
+    setSelectedTenants(newSelectedTentants);
+  }
+
+  const handleDisableSearch = () => {
+    setSearchedTenants(allTenants);
+    setIsSearchActive(false);
+    setDisplayTenants(getDisplayTenants(allTenants, showHoused, showArchived));
+  }
+
+  const handleSearchOutput = (output, isTrue) => {
+    setSearchedTenants(output);
+    setIsSearchActive(isTrue);
+    setDisplayTenants(getDisplayTenants(output, showHoused, showArchived))
+
+  }
+
+  const handleToggleArchived = () => {
+    const newShowArchived = !showArchived;
+    setShowArchived(newShowArchived);
+
+    if (isSearchActive) {
+      setDisplayTenants(getDisplayTenants(searchedTenants, showHoused, !showArchived));
+    } else {
+      setDisplayTenants(getDisplayTenants(allTenants, showHoused, !showArchived));
+    }
+  }
+
+  const toggleArchiveModal = () => {
+    const newShowArchiveModal = !showArchiveModal;
+    setShowArchiveModal(newShowArchiveModal)
```

Comment:
Can we simplify this with just
`setShowArchiveModal(!showArchiveModal);`
?

Reply:
I believe I tried that, and it didn't work.  I pulled the `newShowArchiveModal = !showArchiveModal` from somewhere else.  I might have found a cleaner solution instead.  I'm having trouble with the backend at the moment, so I can't test it, but I might try `setShowArchiveModal(showArchiveModal => !showArchiveModal)`

Reply:
I just tested `setShowArchiveModal(!showArchiveModal)` and it seems to be working.  Not sure where I got the notion it wouldn't.